pkg/operator/ceph/controller/spec.go

```diff
@@ -74,23 +76,40 @@ var logger = capnslog.NewPackageLogger("github.com/rook/rook", "ceph-spec")
 var (
 	cronLogRotate = `
 CEPH_CLIENT_ID=%s
-PERIODICITY=%s
-LOG_ROTATE_CEPH_FILE=/etc/logrotate.d/ceph
+PERIODICITY_IN_HOUR=%s
+LOG_ROTATE_CEPH_FOLDER=/etc/logrotate.d/ceph
+DAEMON_LOG_FILE=/var/log/ceph/$CEPH_CLIENT_ID.log
+LOG_MAX_SIZE=%s
 
-if [ -z "$PERIODICITY" ]; then
-	PERIODICITY=24h
-fi
+LAST_ROTATION_TIME_IN_SEC=$(echo $SECONDS)
 
 # edit the logrotate file to only rotate a specific daemon log
 # otherwise we will logrotate log files without reloading certain daemons
 # this might happen when multiple daemons run on the same machine
-sed -i "s|*.log|$CEPH_CLIENT_ID.log|" "$LOG_ROTATE_CEPH_FILE"
+sed -i "s|*.log|$CEPH_CLIENT_ID.log|" "$LOG_ROTATE_CEPH_FOLDER"
+
+# append LOG_MAX_SIZE at line 3
+sed --in-place "3i size $LOG_MAX_SIZE" "$LOG_ROTATE_CEPH_FOLDER"
 
 while true; do
-	sleep "$PERIODICITY"
-	echo "starting log rotation"
-	logrotate --verbose --force "$LOG_ROTATE_CEPH_FILE"
-	echo "I am going to sleep now, see you in $PERIODICITY"
+
+	logrotate --verbose "$LOG_ROTATE_CEPH_FOLDER"
+
+	CURRENT_TIME_IN_SEC=$(echo $SECONDS)
+	TIME_DIFF_IN_HOUR=$(((LAST_ROTATION_TIME_IN_SEC - CURRENT_TIME_IN_SEC) / 3600))
+	if [ "$TIME_DIFF_IN_HOUR" -lt 0 ]; then
+		TIME_DIFF_IN_HOUR=$((-TIME_DIFF_IN_HOUR))
+	fi
+
+	if [[ $TIME_DIFF_IN_HOUR -ge $PERIODICITY_IN_HOUR ]]; then
+		echo "starting log rotation"
+		logrotate --verbose --force "$LOG_ROTATE_CEPH_FOLDER"
+		LAST_ROTATION_TIME_IN_SEC=$(echo $SECONDS)
+	fi
+
+	echo "Size of file $DAEMON_LOG_FILE to rotate: $(ls -l --human-readable $DAEMON_LOG_FILE | awk '{print $5}')"
```

Comment:
Not every call to logrotate will cause a rotation, so hopefully the output of the logrotate command will just capture it.